assets/src/js/Components/ArchivePicker.js

```diff
@@ -1,151 +1,115 @@
-      this.setState( {
-        currentPage: pageIndex,
-        images: [
-          ...( newSearch ? [] : this.state.images ),
-          ...withPageLabel
-        ],
-        searchText: searchedText,
-      } );
-    } catch ( error ) {
-      this.setState( { error } );
-    } finally {
-      this.setState( { loading: false } );
-    }
-  }
+  const toggleMultiSelection = target => setSelectedIds(
+    selectedIds.includes(target.id)
+      ? selectedIds.filter(id => id !== target.id)
+      : [...selectedIds, target.id]
+  );
 
-  async includeInWp( ids ) {
-    try {
-      this.setState( { processingImages: true } );
-      const updatedImages = await apiFetch( {
-        method: 'POST',
-        path: '/planet4/v1/image-archive/transfer',
-        data: {
-          ids: ids,
-          use_original_language: false,
-        }
-      } );
-      this.updateFromUploadedResponse( updatedImages );
-    } catch ( e ) {
-      console.log( e );
-      this.setState( { processingError: e } );
-    } finally {
-      this.setState( { processingImages: false } );
-    }
-  }
+  useEffect(() => {
+    loadPage();
+  }, []);
 
-  updateFromUploadedResponse( updatedImages ) {
-    const newImages = this.state.images.map( stateImage => {
-      const updated = updatedImages.find( updatedImage => updatedImage.id === stateImage.id );
-      if ( updated ) {
-        return updated;
-      }
-      return stateImage;
-    } );
-    this.setState( {
-      images: newImages,
-    } );
-  }
+  const selectedImages = getSelectedImages(selectedIds);
 
-  async search() {
-    await this.loadNextPage( true );
-  }
-
-  render() {
-    const {
-      loading,
-      error,
-      images,
-    } = this.state;
-
-    return <Fragment>
-      <form
-        className={ 'archive-picker-search' }
-        onSubmit={ event => {
-          event.preventDefault();
-          this.search();
-        } }
-        onChange={ event => this.setState( { enteredSearch: event.target.value } ) }
+  return <Fragment>
+    <form
+      className={'archive-picker-search'}
+      onSubmit={async event => {
+        event.preventDefault();
+        await loadPage(1, enteredSearch);
+        setCurrentPage(1);
+        setSearchText(enteredSearch);
+      }}
+      onChange={event => setEnteredSearch(event.target.value)}
+    >
+      <input
+        type={'text'}
+        disabled={loading}
+      />
+      <input type={'submit'} value={__('Search', 'planet4-master-theme-backend')} />
+    </form>
+    {loading && (
+      <div className={'archive-picker-loading'}> loading...</div>
+    )}
+    {!!error && (
+      <div>
+        <h3>API error:</h3>
+        <p> {error.message} </p>
+      </div>
+    )}
+    <div className={'image-picker'}>
+      <ul
+        className={'picker-list'}
+        onScroll={async (event) => {
+          if (isNearScrollEnd(event)) {
+            await loadPage(currentPage + 1, searchText);
+            setCurrentPage(currentPage + 1);
```

Comment:
If we only use the `setCurrentPage` call here, and pass `currentPage` as a dependency [here](https://github.com/greenpeace/planet4-master-theme/pull/1196/files?file-filters%5B%5D=.js&file-filters%5B%5D=No+extension&hide-deleted-files=true&w=1#diff-867d658bf27f0e63a343817562d12adeR47), it should do the same thing.

Reply:
We also would need to add the search text there, but that would make sense regardless. We might later keep track of the search text in the url and apply the search filter on the initial load.

Reply:
Updated! 🙂